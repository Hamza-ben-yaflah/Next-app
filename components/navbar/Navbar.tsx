import React from "react";
import { Layout, Button, Input } from "antd";
import styles from "./Navbar.module.css";
import { SearchOutlined } from "@ant-design/icons";
import Image from "next/image";
import car from "./car.jpg";

const { Header } = Layout;
const { Search } = Input;
function Navbar() {
  return (
    <div className={styles.layout}>
      <Header className={styles.header}>
        <div className={styles.imageContainer}>
          <Image
            src={car}
            alt="car image"
            width={120}
            height={70}
            className={styles.imgCar}
          />
        </div>
        <Search placeholder="Search" style={{ width: 200 }} size="large" />

        <Button type="primary" className={styles.btn} size="large">
          Log in
        </Button>
        <Button type="primary" className={styles.btn} size="large">
          Sign up
        </Button>
      </Header>
    </div>
  );
}

export default Navbar;
