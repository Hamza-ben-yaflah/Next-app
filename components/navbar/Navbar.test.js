import { render, screen } from "@testing-library/react";
import Navbar from "./Navbar";
import renderer from "react-test-renderer";

test("render the navbar", () => {
  const nav = renderer.create(<Navbar />).toJSON();
  expect(nav).toMatchSnapshot();
});

test("render the navbar", () => {
  render(<Navbar />);
  const nar = screen.getByRole("banner");
  expect(nar).toBeInTheDocument();
});
