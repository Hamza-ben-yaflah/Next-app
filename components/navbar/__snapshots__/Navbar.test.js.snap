// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the navbar 1`] = `
<header
  className="ant-layout-header header"
  style={
    Object {
      "position": "fixed",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <div
    className="imageContainer"
  >
    <a
      className="ant-btn ant-btn-link ant-btn-lg btn"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <span
        aria-label="home"
        className="anticon anticon-home"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="home"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </span>
      <span>
        Home
      </span>
    </a>
    <button
      className="ant-btn ant-btn-link ant-btn-lg btn"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-label="question-circle"
        className="anticon anticon-question-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="question-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
          />
        </svg>
      </span>
      <span>
        About
      </span>
    </button>
    <button
      className="ant-btn ant-btn-link ant-btn-lg btn"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-label="customer-service"
        className="anticon anticon-customer-service"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="customer-service"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 128c-212.1 0-384 171.9-384 384v360c0 13.3 10.7 24 24 24h184c35.3 0 64-28.7 64-64V624c0-35.3-28.7-64-64-64H200v-48c0-172.3 139.7-312 312-312s312 139.7 312 312v48H688c-35.3 0-64 28.7-64 64v208c0 35.3 28.7 64 64 64h184c13.3 0 24-10.7 24-24V512c0-212.1-171.9-384-384-384zM328 632v192H200V632h128zm496 192H696V632h128v192z"
          />
        </svg>
      </span>
      <span>
        Service
      </span>
    </button>
    <button
      className="ant-btn ant-btn-link ant-btn-lg btn"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-label="snippets"
        className="anticon anticon-snippets"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="snippets"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M832 112H724V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H500V72c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v40H320c-17.7 0-32 14.3-32 32v120h-96c-17.7 0-32 14.3-32 32v632c0 17.7 14.3 32 32 32h512c17.7 0 32-14.3 32-32v-96h96c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM664 888H232V336h218v174c0 22.1 17.9 40 40 40h174v338zm0-402H514V336h.2L664 485.8v.2zm128 274h-56V456L544 264H360v-80h68v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h152v32c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-32h68v576z"
          />
        </svg>
      </span>
      <a
        href="/Blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
    </button>
  </div>
  <div
    className="loginContainer"
  >
    <span
      className="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <span
        className="ant-input-wrapper ant-input-group"
      >
        <input
          className="ant-input ant-input-lg"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          style={null}
          type="text"
          value=""
        />
        <span
          className="ant-input-group-addon"
        >
          <button
            className="ant-btn ant-btn-lg ant-btn-icon-only ant-input-search-button"
            onClick={[Function]}
            onMouseDown={[Function]}
            type="button"
          >
            <span
              aria-label="search"
              className="anticon anticon-search"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </span>
    <button
      className="ant-btn ant-btn-primary ant-btn-lg btn"
      onClick={[Function]}
      type="button"
    >
      <a
        href="/LoginRegister"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Log in
      </a>
    </button>
    <button
      className="ant-btn ant-btn-primary ant-btn-lg btn"
      onClick={[Function]}
      type="button"
    >
      <a
        href="/RegisterPage"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        log out
      </a>
    </button>
  </div>
</header>
`;
